Amazon.QLDB.Driver.IntegrationTests/AsyncStatementExecutionTests.cs

```diff
@@ -0,0 +1,1053 @@
+
+            // When.
+            var query = $"INSERT INTO {Constants.TableName} ?";
+            await qldbDriver.Execute(async txn => await txn.Execute(query, ionStruct));
+
+            // Then.
+            var searchQuery = $@"SELECT VALUE {Constants.ColumnName} FROM {Constants.TableName}
+                                 WHERE {Constants.ColumnName} = '{Constants.SingleDocumentValue}'";
+            var value = await qldbDriver.Execute(async txn =>
+            {
+                var result = await txn.Execute(searchQuery);
+
+                string value = "";
+                await foreach (var row in result)
+                {
+                    value = row.StringValue;
+                }
+                return value;
+            });
+            Assert.AreEqual(Constants.SingleDocumentValue, value);
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(BadRequestException))]
+        public async Task Execute_DeleteTableThatDoesntExist_ThrowsBadRequestException()
+        {
+            // Given.
+            var query = "DELETE FROM NonExistentTable";
+
+            // When.
+            await qldbDriver.Execute(async txn => await txn.Execute(query));
+        }
+
+        [TestMethod]
+        public async Task Execute_ExecutionMetrics()
+        {
+            await qldbDriver.Execute(async txn =>
+            {
+                var insertQuery = String.Format("INSERT INTO {0} << {{'col': 1}}, {{'col': 2}}, {{'col': 3}} >>",
+                    Constants.TableName);
+                await txn.Execute(insertQuery);
+            });
+
+            // Given
+            var selectQuery = String.Format("SELECT * FROM {0} as a, {0} as b, {0} as c, {0} as d, {0} as e, {0} as f",
+                Constants.TableName);
+
+            // When
+            await qldbDriver.Execute(async txn =>
+            {
+                var result = await txn.Execute(selectQuery);
+
+                await foreach (IIonValue row in result)
+                {
+                    var ioUsage = result.GetConsumedIOs();
+                    var timingInfo = result.GetTimingInformation();
+
+                    Assert.IsNotNull(ioUsage);
+                    Assert.IsNotNull(timingInfo);
+                    Assert.IsTrue(ioUsage?.ReadIOs > 0);
+                    Assert.IsTrue(timingInfo?.ProcessingTimeMilliseconds > 0);
+                }
+            });
+
+            // When
+            var result = await qldbDriver.Execute(async txn =>
+            {
+                return await txn.Execute(selectQuery);
+            });
+
+            var ioUsage = result.GetConsumedIOs();
+            var timingInfo = result.GetTimingInformation();
+
+            Assert.IsNotNull(ioUsage);
+            Assert.IsNotNull(timingInfo);
+            Assert.AreEqual(1092, ioUsage?.ReadIOs);
+            Assert.IsTrue(timingInfo?.ProcessingTimeMilliseconds > 0);
+        }
+
+        public static IEnumerable<object[]> CreateIonValues()
```

Comment:
This can be moved to IntegrationTestBase so async and non-async tests can use this.

As far as I can see, theres no async functionality in this function

Reply:
Addressed